staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -237,18 +189,34 @@ func (gmf *GroupMetaFactory) Enable(m *registered.APIRegistrationManager, scheme
 		return err
 	}
 	return nil
+
 }
 
-// RegisterAndEnable is provided only to allow this code to get added in multiple steps.
-// It's really bad that this is called in init() methods, but supporting this
-// temporarily lets us do the change incrementally.
-func (gmf *GroupMetaFactory) RegisterAndEnable(registry *registered.APIRegistrationManager, scheme *runtime.Scheme) error {
-	if err := gmf.Register(registry); err != nil {
-		return err
+func (gmf *GroupMetaFactory) newRESTMapper(scheme *runtime.Scheme, externalVersions []schema.GroupVersion, groupMeta *apimachinery.GroupMeta) meta.RESTMapper {
+	// the list of kinds that are scoped at the root of the api hierarchy
+	// if a kind is not enumerated here, it is assumed to have a namespace scope
+	rootScoped := sets.NewString()
+	if gmf.GroupArgs.RootScopedKinds != nil {
+		rootScoped = gmf.GroupArgs.RootScopedKinds
```

Comment:
any reason not to do this immediately? Why the if clause?

Reply:
> any reason not to do this immediately? Why the if clause?

Daniel doc'd it as possibly nil and I don't know what else falls over on nils for it yet.  I think that restmapping is going away from apiregistration, since you don't logically know it at registration time.